RELEASES.md

```diff
@@ -1442,7 +1442,7 @@ Compatibility Notes
 - [Mixing Option and Result via `?` is no longer permitted in closures for inferred types.][86831]
 - [Previously unsound code is no longer permitted where different constructors in branches
   could require different lifetimes.][85574]
-- As previously mentioned the [`std::arch` instrinsics now uses stricter const checking][83278]
+- As previously mentioned the [`std::arch` intrinsic now uses stricter const checking][83278]
```

Comment:
`std::arch` isn't an intrinsic, it has intrinsics. The plural is correct here, but it was not previously spelled correctly